### DISCUSSION TITLE
webpack 4 hardcode md4 as hash function which not able to use on node 17 or above

### DISCUSSION THREAD

**Discussion by duc2019** [webpack/webpack#16733] [Q&A] [answered]
https://github.com/webpack/webpack/blob/webpack-4/lib/optimize/SplitChunksPlugin.js#L25

Seriously???

```
const hashFilename = name => {
	return crypto
		.createHash("md4")
		.update(name)
		.digest("hex")
		.slice(0, 8);
};
```

You guys provide output.hashFunction option but it is useless.

- I'm using nuxtjs, which only support webpack 4: https://github.com/nuxt/nuxt/issues/9268
- webpack4 hardcode md4 in hash filename
- md4 is forbidden on node 17 and above

PLEASE provide a option for webpack4 to use custom hash function for hashFilename.

Related stackoverflow question (anwsers are useless).
https://stackoverflow.com/questions/69692842/error-message-error0308010cdigital-envelope-routinesunsupported

**Comment by alexander-akait** [comment, marked as answer]
Sorry, but yes, webpck v4 is outdated, please ask for migrate on webpack v5, there is an option for this, anyway you can monkey patch Node.js runtime and add md4 support, feel free to feedback
